test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/IdentityHashMapTest.scala

```diff
@@ -0,0 +1,397 @@
+    assertTrue(entrySet.remove(SIE(testObj(5), null)))
+
+    mp.put(testObj(1), testObj(11))
+    mp.put(testObj(3), testObj(33))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3)))
+
+    entrySet.removeAll(ju.Arrays.asList(SIE(testObj(1), TestObj(11)),
+        SIE(testObj(2), testObj(22)), SIE(testObj(3), testObj(44)),
+        TestObj(3), 33))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertFalse(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3)))
+
+    mp.put(testObj(2), testObj(22))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3)))
+
+    entrySet.retainAll(ju.Arrays.asList(SIE(testObj(1), testObj(11)),
+        SIE(testObj(2), testObj(22)), SIE(testObj(3), TestObj(33)),
+        testObj(3), 42))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2)))
+    assertFalse(mp.containsKey(testObj(3)))
+  }
+
+  @Test def testIdentityKeySetIsViewForQueriesWithCustomObjects(): Unit = {
+    val mp = factory.empty[TestObj, TestObj]
+    mp.put(testObj(1), TestObj(11))
+    mp.put(testObj(2), TestObj(22))
+    val keySet = mp.keySet()
+
+    assertTrue(keySet.contains(testObj(1)))
+    assertFalse(keySet.contains(TestObj(1)))
+    assertTrue(keySet.contains(testObj(2)))
+    assertFalse(keySet.contains(TestObj(2)))
+    assertFalse(keySet.contains(testObj(3)))
+
+    mp.put(testObj(3), TestObj(33))
+
+    assertTrue(keySet.contains(testObj(3)))
+    assertFalse(keySet.contains(TestObj(3)))
+
+    val coll1 = ju.Arrays.asList(testObj(1), testObj(2), testObj(3))
+    val coll1a = ju.Arrays.asList(testObj(1), testObj(2), TestObj(3))
+    assertTrue(keySet.containsAll(coll1))
+    assertFalse(keySet.containsAll(coll1a))
+  }
+
+  @Test def testIdentityKeySetIsViewForRemoveWithCustomObjects(): Unit = {
+    assumeFalseIdentityHashMapViews()
+
+    val mp = factory.empty[TestObj, TestObj]
+    mp.put(testObj(1), TestObj(11))
+    mp.put(testObj(2), TestObj(22))
+    val keySet = mp.keySet()
+
+    assertFalse(keySet.isEmpty)
+    assertFalse(mp.isEmpty)
+
+    keySet.clear()
+
+    assertTrue(keySet.isEmpty)
+
+    assertTrue(mp.isEmpty)
+
+    mp.put(testObj(1), TestObj(11))
+    mp.put(testObj(2), TestObj(22))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    keySet.remove(testObj(1))
+    assertFalse(mp.containsKey(testObj(1)))
+
+    mp.put(testObj(1), TestObj(11))
+    mp.put(testObj(3), TestObj(33))
+
+    assertTrue(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2)))
+    assertTrue(mp.containsKey(testObj(3)))
+    assertFalse(mp.containsKey(TestObj(3)))
+
+    val coll1 = ju.Arrays.asList(testObj(1), TestObj(2), testObj(3), testObj(5))
+    keySet.removeAll(coll1)
+
+    assertFalse(mp.containsKey(testObj(1)))
+    assertTrue(mp.containsKey(testObj(2))) // fails JVM
```

Comment:
If only a few individual asserts fail on the JVM, please guard only those with a `if (!executingInJVM)`, rather than assuming away the entire `@Test def`. The same applies to the other places where you put `// fails JVM`.

Reply:
Much better, comments removed too. No sense in ignoring the entire test for only those items where this collection is better than the JVM version. 😄